pkg/containers/images.go

```diff
@@ -0,0 +1,338 @@
+	return &Client{
+		Kube: kube.GetConfigInstance(kubeContext),
+	}
+}
+
+// Find is the primary function for this package that returns the results of images found in the cluster and whether they are out of date or not
+func (c *Client) Find() (Results, error) {
+	clusterImages, err := c.getContainerImages()
+	if err != nil {
+		return Results{}, err
+	}
+	if len(clusterImages) == 0 {
+		return Results{}, fmt.Errorf("no container images found in cluster")
+	}
+
+	images := make([]*Image, len(clusterImages))
+	errored := make([]*ErroredImage, 0)
+	g := new(errgroup.Group)
+	for i, fullName := range clusterImages {
+		i, fullName := i, fullName
+		image, err := newImage(fullName)
+		if err != nil {
+			errored = append(errored, &ErroredImage{
+				Image: fullName,
+				Err:   err.Error(),
+			})
+			images[i] = nil
+			continue
+		}
+		klog.V(5).Infof("Getting tags for %s", image.Name)
+		g.Go(func() error {
+			err := image.getTags()
+			if err == nil {
+				images[i] = image
+			}
+			return err
+		})
+	}
+	if err := g.Wait(); err != nil {
+		fmt.Println("Got an error when getting remote tags:", err)
+		// return Results{}, err
+	}
+	for _, image := range images {
+		if image == nil {
+			continue
+		}
+		image.parseTags()
+		err := image.populateNewest()
+		if err != nil {
+			return Results{}, err
+		}
+	}
+	return Results{
+		Images:    images,
+		ErrImages: errored,
+	}, nil
+}
+
+// getContainerImages fetches all pods and returns a slice of container images
+func (c *Client) getContainerImages() ([]string, error) {
+	klog.V(3).Infof("Getting container images from pods")
+
+	k := c.Kube.Client
+	pods, err := k.CoreV1().Pods("").List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return nil, errors.Wrap(err, "getting all pods")
+	}
+
+	imagesFound := make([]string, 0)
+	for _, pod := range pods.Items {
+		if len(pod.Spec.InitContainers) > 0 {
+			for _, container := range pod.Spec.InitContainers {
+				if container.Image != "" {
+					imagesFound = append(imagesFound, container.Image)
+				}
+			}
+		}
+		for _, container := range pod.Spec.Containers {
+			if container.Image != "" {
+				imagesFound = append(imagesFound, container.Image)
+			}
+		}
+	}
+	imagesFound = removeDuplicateStr(imagesFound)
+	return imagesFound, nil
+}
+
+func removeDuplicateStr(strSlice []string) []string {
+	allKeys := make(map[string]bool)
+	list := []string{}
+	for _, item := range strSlice {
+		if _, value := allKeys[item]; !value {
+			allKeys[item] = true
+			list = append(list, item)
+		}
+	}
+	return list
+}
+
+func newImage(fullImageTag string) (*Image, error) {
+	klog.V(3).Infof("Creating image object for %s", fullImageTag)
+
+	var (
+		err     error
+		repo    = ""
+		currTag = "latest"
+		image   = new(Image)
+	)
+
+	splitString := strings.Split(fullImageTag, ":")
+	if len(splitString) > 0 {
+		repo = splitString[0]
+		if len(splitString) > 1 {
+			currTag = splitString[1]
+		}
+	}
+
+	re, err := regexp.Compile(`^v[0-9]+.*$`)
```

Comment:
what happens if the tag is `latest`?

Reply:
This particular regex is only looking for prefix of a valid semver. It's ok if it doesn't match here, but to answer your question, I do think there is an issue with using `latest` tag or maybe any tag that is only a string. I'll look into it, thanks!

Reply:
Ok, so if the current tag is `latest` it would have never shown up in the output, but I have fixed that now by allowing the `--include-all/-a` flag make that show up. The reason it wasn't showing up before is that if we decided the newest image was already deployed it would never show in the output. Now it should work.